src/metabase/channel/template/handlebars_helper.clj

```diff
@@ -0,0 +1,129 @@
+(ns metabase.channel.template.handlebars-helper
+  (:refer-clojure :exclude [hash])
+  (:require
+   [java-time.api :as t]
+   [metabase.util.date-2 :as u.date]
+   [metabase.util.urls :as urls])
+  (:import
+   (com.github.jknack.handlebars
+    Options Handlebars Helper Handlebars$SafeString)))
+
+(set! *warn-on-reflection* true)
+
+(defn option-param
+  "Get a parameter from the options by index."
+  ([^Options option idx]
+   (.param option idx))
+  ([^Options option idx default]
+   (.param option idx default)))
+
+(defn option-hash
+  "Get a parameter from the options by key."
+  ([^Options option key]
+   (.hash option key))
+  ([^Options option key default]
+   (.hash option key default)))
+
+(defn option-block-body
+  "Get the block body."
+  [^Options option]
+  (.fn option))
+
+(defn option-else-block
+  "Get the else block."
+  [^Options option]
+  (.inverse option))
+
+(defn safe-str
+  "A wrapper of Handlebars.SafeString"
+  [& text]
+  (Handlebars$SafeString. (apply str text)))
+
+(defmacro defhelper
+  "Define a helper function."
+  [helper-name description argvec & body]
+  (let [helper-fn-name (symbol (str helper-name \*))
+        description#   description]
+    `(do
+       (defn ~helper-fn-name
+         ~description#
+         ~argvec
+         ~@body)
+       (def ~(vary-meta helper-name
+                        merge
+                        {:doc        description
+                         :is-helper? true})
+         ~description#
+         (reify Helper (apply [_# context# option#]
+                         (~helper-fn-name context# option#)))))))
+
+(defn register-helper
+  "Register a helper."
+  [^Handlebars hbs ^String name ^Helper helper]
+  (.registerHelper hbs name helper))
+
+(defn register-helpers
+  "Register a list of helpers."
+  [^Handlebars hbs helpers]
+  (doseq [[name helper] helpers]
+    (register-helper hbs name helper)))
+
+;;---------------- Predefined helpers ----------------;;
+
+;; Generics
+(defhelper equals
+  "Return true if two values are equal.
+
+  {{if (equals product.name \"Hot Dog\")}}
+    Hot Dog
+  {{else}}
+    Not Hot Dog
+  {{/if}}"
+  [arg options]
+  (let [x arg
+        y (option-param options 0)]
+    (= x y)))
+
+(defhelper format-date
```

Comment:
what's the best naming convention to use here? Are dashes ok, or should we do 🐫? 🐍 ?

Reply:
I'm happy with kebab. If we end up making this CLJC for JS as well then Camel would be weird anyway.